pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/FunctionRuntimeManager.java

```diff
@@ -452,25 +444,28 @@ public synchronized void restartFunctionInstances(String tenant, String namespac
                         }
                         continue;
                     }
-
-                    ComponentType componentType = assignment.getInstance().getFunctionMetaData().getFunctionDetails().getComponentType();
-
-                    if (ComponentType.SOURCE == componentType) {
-                        this.functionAdmin.sources().restartSource(tenant, namespace, functionName,
-                            assignment.getInstance().getInstanceId());
-                    } else if (ComponentType.SINK == componentType) {
-                        this.functionAdmin.sinks().restartSink(tenant, namespace, functionName,
-                            assignment.getInstance().getInstanceId());
-                    } else {
-                        this.functionAdmin.functions().restartFunction(tenant, namespace, functionName,
-                            assignment.getInstance().getInstanceId());
-                    }
+                    restartFunctionByPulsarAdmin(assignment, tenant, namespace, functionName);
                 }
             }
         }
         return;
     }
 
+    private void restartFunctionByPulsarAdmin(Assignment assignment, String tenant, String namespace, String functionName)
```

Comment:
what about
`restartFunctionUsingPulsarAdmin` ?

Not a big deal for me, just a suggestion

Reply:
I think your proposal is better. Thanks.